tools/http2_interop/http2interop.go

```diff
@@ -195,21 +191,22 @@ func testUnknownFrameType(ctx *HTTP2InteropCtx) error {
 }
 
 func testShortPreface(ctx *HTTP2InteropCtx, prefacePrefix string) error {
-	c, err := connect(ctx)
+	conn, err := connect(ctx)
 	if err != nil {
 		return err
 	}
-	defer c.Close()
+	defer conn.Close()
+	conn.SetDeadline(time.Now().Add(defaultTimeout))
 
-	if _, err := c.Write([]byte(prefacePrefix)); err != nil {
+	if _, err := conn.Write([]byte(prefacePrefix)); err != nil {
 		return err
 	}
 
-	buf := make([]byte, 256)
-	for ; err == nil; _, err = c.Read(buf) {
+	if _, err := expectGoAwaySoon(conn); err != nil {
```

Comment:
expectGoAway is for when the very first frame should be a Go away.  This is for things like bad TLS configuration.   Sometimes it isn't feasible for the TLS terminator to teardown the connection, so the spec allows sending a goaway frame with inadequate security before any other frames.

However, most servers, upon getting a good TLS connection, will immediately send their preface before actually receiving any bytes from the client, in order to speed up the handshake.  In this test, I am intentionally sending a bad preface to the server, which means the server preface may already be on the wire.  This code expects a few valid frames (like settings, continuations, and window updates) before getting a goaway frame from the server.